src/app/DefaultAttributePersistenceProvider.cpp

```diff
@@ -38,16 +38,16 @@ CHIP_ERROR DefaultAttributePersistenceProvider::WriteValue(const ConcreteAttribu
         aValue.data(), static_cast<uint16_t>(aValue.size()));
 }
 
-CHIP_ERROR DefaultAttributePersistenceProvider::ReadValue(const ConcreteAttributePath & aPath,
-                                                          const EmberAfAttributeMetadata * aMetadata, MutableByteSpan & aValue)
+CHIP_ERROR DefaultAttributePersistenceProvider::ReadValue(const ConcreteAttributePath & aPath, EmberAfAttributeType aType,
+                                                          uint16_t aSize, MutableByteSpan & aValue)
```

Comment:
Suggest size_t instead of uint16_t

Reply:
I have used `uint16_t` so that it matches with the `size` member of `EmberAfAttributeMetadata`.

Reply:
Please use size_t and just case. EmberAfAttributeMetadata is very unfortunately opinionated about uint16_t when a size_t is the correct canonical type.